### PR TITLE
item_collection refactor

### LINKED ISSUE #424: General refactoring
### Current Behaviour:
A lot of functions have many if-else branches and the procedures implemented by these branches have too much code in them. This makes readability and testability hard.

For example the branch in this [chunk of code](https://github.com/HTTP-APIs/hydrus/blob/42205ab4d4984f591310b19f3b5d6c1610e95e4d/hydrus/resources.py#L108) should be ported into a function like:
```
    if checkClassOp(class_path, "GET"):
        items_get_check_support() 
    abort(405)
```
Into a dedicated module called `itemshelpers.py` the above functions should be defined by incorporating the current code:
```
def items_get_check_support(...):
    """ Check if class_type supports GET operation
            try:
                # Try getting the Item based on ID and Class type
                response = crud.get(
                    id_,
                    class_type,
                    api_name=get_api_name(),
                    session=get_session())

                response = finalize_response(class_path, response)
                return set_response_headers(
                    jsonify(hydrafy(response, path=path)))

            except (ClassNotFound, InstanceNotFound) as e:
                error = e.get_HTTP()
                return set_response_headers(jsonify(error.generate()), status_code=error.code)
```
Each of these helpers functions should be tested separately.

### Expected Behaviour:
All functions should perform well-scoped units of work and have their own tests.
Handlers in endpoints classes should be much more concise, all the units of work for each branch should be moved into helpers functions. 

Main functions should look something like this:
```
def post(...):
    """ Docstring """
    ... some code ...
    if condition_1:
        run_helper_1(...)
    elif condition_2:
        run_helper_2(...)
    else:
        run_helper_3(...)
    ... rest of the code ...
```

### PR DESCRIPTION
**Pull request by thrishik7**
Fixes #424 

### Checklist
- [ ] My branch is up-to-date with upstream/develop branch.
- [ ] Everything works and tested for Python 3.5.2 and above.

### Description
refactor class itemcollection

### CODE REVIEW

File: hydrus/itemhelpers.py [hunk 1]
```diff
@@ -155,3 +170,98 @@ def items_delete_check_support(id_, class_type, path):
         error = e.get_HTTP()
         return set_response_headers(jsonify(error.generate()),
                                     status_code=error.code)
+
+
+def itemsCollection_get_support(collection, class_path, path, search_params):
+            try:
```
Comment (xadahiya) [inline hydrus/itemhelpers.py]: Inconsistent Indentation

File: hydrus/itemhelpers.py [hunk 2]
```diff
@@ -155,3 +170,98 @@ def items_delete_check_support(id_, class_type, path):
...
+
+                return set_response_headers(jsonify(hydrafy(response, path=path)))
+
+            except (ClassNotFound, PageNotFound, InvalidSearchParameter, OffsetOutOfRange) as e:
+                error = e.get_HTTP()
+                return set_response_headers(jsonify(error.generate()), status_code=error.code)
+
+
+def itemsCollection_post_support(object_, link_props, path):
+    try:
```
Comment (xadahiya) [inline hydrus/itemhelpers.py]: Please add doctrings to all your functions.

Comment (xadahiya) [reply hydrus/itemhelpers.py]: Doctrings should be in a format similar to https://github.com/HTTP-APIs/hydrus/blob/37af5f3be07b5b21e133c40e3586f58859f38094/hydrus/helpers.py

File: hydrus/resources.py [hunk 3]
```diff
@@ -64,7 +64,11 @@
     items_get_check_support,
     items_post_check_support,
```
Comment (xadahiya) [inline hydrus/resources.py]: Please remove all the redundant imports that are not being used here.

**Comment by thrishik7** [comment]
> Great effort! However, you need to make a couple of changes before we can proceed.

okay @xadahiya , I will work on it

**Comment by thrishik7** [comment]
@xadahiya  I have made the changes. Please let me know if I have to add few more changes

**Comment by xadahiya** [comment]
All docstings still don't have consistent format.

**Comment by thrishik7** [comment]
> All docstings still don't have consistent format.

@xadahiya okay, I will work on it.

**Comment by thrishik7** [comment]
@xadahiya can you please point out what to change for anyone docstring. So that I would apply the same for the rest

**Comment by xadahiya** [comment]
See `itemsCollection_get_support` for example docstring

**Comment by thrishik7** [comment]
> See `itemsCollection_get_support` for example docstring
@xadahiya  should I refer `itemsCollection_get_support` and change other docstrings??

**Comment by xadahiya** [comment]
All docstrings should have a consistent format like this

**Pull request closed without merging**